build-logic/jvm/src/main/kotlin/gradlebuild.unittest-and-compile.gradle.kts

```diff
@@ -183,12 +186,27 @@ fun Test.jvmVersionForTest(): JavaLanguageVersion {
     return JavaLanguageVersion.of(project.testJavaVersion)
 }
 
-fun Test.configureTestQuarantine() {
-    if (project.flakyTestQuarantine.isPresent) {
-        systemProperty("spock.configuration", "FlakyTestQuarantineSpockConfig.groovy")
-        (options as JUnitPlatformOptions).includeTags("org.gradle.test.fixtures.Flaky")
-    } else {
-        (options as JUnitPlatformOptions).excludeTags("org.gradle.test.fixtures.Flaky")
+fun Test.configureSpock() {
+    systemProperty("spock.configuration", "GradleBuildSpockConfig.groovy")
+}
+
+fun Test.configureFlakyTest() {
+    when (project.flakyTestStrategy) {
+        FlakyTestStrategy.INCLUDE -> {}
+        FlakyTestStrategy.EXCLUDE -> {
+            excludeSpockAnnotation("org.gradle.test.fixtures.Flaky")
+            (options as JUnitPlatformOptions).excludeTags("org.gradle.test.fixtures.Flaky")
+        }
+        FlakyTestStrategy.QUARANTINE -> {
+            // This is a very ugly workaround for https://github.com/spockframework/spock/issues/1288
+            // JUnit Platform `includeTags` works before Spock engine, thus exclude ALL spock tests.
+            // For now, we only use `includeTags` for non-spock subproject
+            if (project.name.startsWith("kotlin-dsl")) {
```

Comment:
Is there another option how to detect whether the test task runs Spock tests or not? The `kotlin-dsl` subproject also has Spock test, though no flaky ones I suppose.

Also, does this mean this is currently broken, since we don't include any Spock tests in the flaky test quarantine?

Reply:
Maybe something we could do here is to add some category, like `VintageIntegrationTest` to `AbstractIntegrationSpec`, and then use `includeTags("none() | Flaky")` here, given that the JUnit integration test would inherit (hopefully) the `VintageIntegrationTest` tag from the base class. WDYT?